.github/workflows/style_check.yml

```diff
@@ -5,36 +5,40 @@ concurrency:
   cancel-in-progress: true
 
 on:
-  # Trigger on pull requests.
   pull_request:
 
-  # Trigger on pushes to main
   push:
     branches:
       - "main"
 
-  # Trigger on request.
   workflow_dispatch:
 
+defaults:
+  run:
+    shell: bash -leo pipefail {0}
+
 jobs:
   clang-tidy:
     name: Check C++ Code Style
-    runs-on: ubuntu-latest
-    container:
-      image: glotzerlab/ci:2021.11-clang10_py38
+    runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v4.1.6
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

Comment:
Can we just use the version here instead of the commit hash?

Reply:
We *can*, but we **should not**: 
* https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash
* https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Reply:
Okay, but I have a couple of questions. Should we be doing this for every github action? How will we know to update the hash when a new version of the action is released?

Reply:
Yes, we should ideally use this for every GitHub action to ensure consistent behavior.
Dependabot will update both the hash and comment when new versions are released:
https://github.com/dependabot/dependabot-core/pull/5951